frontend/src/components/editor/Cell.tsx

```diff
@@ -224,6 +234,34 @@ const CellComponent = (
   // Callback to get the editor view.
   const getEditorView = useCallback(() => editorView.current, [editorView]);
 
+  const { openModal } = useImperativeModal();
+
+  const handleName = useEvent(() => {
+    alert("handleName");
+    openModal(
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Rename cell</DialogTitle>
+        </DialogHeader>
+        <div className="flex items-center justify-between">
+          <Label htmlFor="cell-name">Cell name</Label>
+          <NameCellInput
+            placeholder={`cell_${cellId}`}
+            value={name}
+            onKeyDown={(e) => {
+              if (e.key === "Enter") {
+                e.preventDefault();
+                e.stopPropagation();
+                openModal(null);
+              }
+            }}
+            onChange={(newName) => updateCellName({ cellId, name: newName })}
+          />
+        </div>
+      </DialogContent>,
+    );
+  });
```

Comment:
since this is used elsewhere, we can consolidate this:

```ts
export function useHandleRenameCell(updateCellName: CellActions["updateCellName"]) {
const { openModal, closeModal } = useImperativeModal();

  const handleName = useEvent(() => {
    alert("handleName");
    openModal(
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Rename cell</DialogTitle>
        </DialogHeader>
        <div className="flex items-center justify-between">
          <Label htmlFor="cell-name">Cell name</Label>
          <NameCellInput
            placeholder={`cell_${cellId}`}
            value={name}
            onKeyDown={(e) => {
              if (e.key === "Enter") {
                e.preventDefault();
                e.stopPropagation();
                closeModal();
              }
            }}
            onChange={(newName) => updateCellName({ cellId, name: newName })}
          />
        </div>
      </DialogContent>,
    );
  });
}
```

Reply:
Can we put it in this file `frontend/src/hooks/useHandleRenameCell.ts`

Reply:
yep, thats a good place for this